screeps-game-api/src/constants.rs

```diff
@@ -775,10 +821,59 @@ impl ResourceType {
             CatalyzedGhodiumAlkalide => 90,
         }
     }
+
+    pub fn to_string(&self) -> String {
```

Comment:
Argh, I forgot about those!
I don't think having the string instead of the variant is useful. It was only meant to convert to a string before passing the hand to JS.

How about a wrapper around those JS functions that would implement `From<ResourceType> for Value::String` and `TryFrom<Value::String> for ResourceType` using those js methods instead of the usual conversion provided from stdweb?

Reply:
The problem with `TryFrom<Value::String>` is that at that point the string has already been loaded into Rust, so it's equivalent to doing a method like this.

`Value::String` just wraps a rust `String`, not a pointer to a JS string. If we construct it or consume it at all we're already paying the cost of transfering the string between Rust and JS, which is what I want to avoid.